ccm_web/server/src/api/api.controller.ts

```diff
@@ -1,60 +1,61 @@
 import { SessionData } from 'express-session'
 import {
-  Body, Controller, Get, HttpException, Param, ParseIntPipe, Post, Put, Session
+  Body, Controller, Get, HttpException, Param, ParseIntPipe, Post, Put, Session, UseGuards
 } from '@nestjs/common'
-import { ApiBearerAuth } from '@nestjs/swagger'
+import { ApiSecurity } from '@nestjs/swagger'
 
 import { Globals, isAPIErrorData } from './api.interfaces'
 import { APIService } from './api.service'
 import { CourseNameDto } from './dtos/api.course.name.dto'
-import { CanvasCourseBase, CanvasCourseSection } from '../canvas/canvas.interfaces'
 import { CreateSectionsDto } from './dtos/api.create.sections.dto'
+import { JwtAuthGuard } from '../auth/jwt-auth.guard'
+import { CanvasCourseBase, CanvasCourseSection } from '../canvas/canvas.interfaces'
+import { UserDec } from '../user/user.decorator'
+import { User } from '../user/user.model'
 
-@ApiBearerAuth()
+@UseGuards(JwtAuthGuard)
 @Controller('api')
 export class APIController {
   constructor (private readonly apiService: APIService) {}
 
   @Get('globals')
-  getGlobals (@Session() session: SessionData): Globals {
-    return this.apiService.getGlobals(session)
+  getGlobals (@Session() session: SessionData, @UserDec() user: User): Globals {
+    return this.apiService.getGlobals(user, session)
   }
 
   @Get('course/:id/sections')
   async getCourseSections (
-    @Param('id', ParseIntPipe) courseId: number, @Session() session: SessionData
+    @Param('id', ParseIntPipe) courseId: number, @UserDec() user: User
   ): Promise<CanvasCourseSection[]> {
-    const { userLoginId } = session.data
-    const result = await this.apiService.getCourseSections(userLoginId, courseId)
+    const result = await this.apiService.getCourseSections(user, courseId)
     if (isAPIErrorData(result)) throw new HttpException(result, result.statusCode)
     return result
   }
 
   @Get('course/:id/name')
   async getCourseName (
-    @Param('id', ParseIntPipe) courseId: number, @Session() session: SessionData
+    @Param('id', ParseIntPipe) courseId: number, @UserDec() user: User
   ): Promise<CanvasCourseBase> {
-    const { userLoginId } = session.data
-    const result = await this.apiService.getCourseName(userLoginId, courseId)
+    const result = await this.apiService.getCourseName(user, courseId)
     if (isAPIErrorData(result)) throw new HttpException(result, result.statusCode)
     return result
   }
 
+  @ApiSecurity('CSRF-Token')
```

Comment:
Why don't the various `GET` API entries also use CSRF security?

Reply:
My understanding is that `GET` requests are not vulnerable in the same way to CSRF attacks. I think these attacks basically try to embed mutating requests to your application in a form on another site, so that it can take advantage of the JWT you have in a cookie that will get sent automatically. However, because of how that works, the other site doesn't have easy access to the response. 

"CSRF attacks target functionality that causes a state change on the server, such as changing the victim’s email address or password, or purchasing something. Forcing the victim to retrieve data doesn’t benefit an attacker because the attacker doesn’t receive the response, the victim does. As such, CSRF attacks target state-changing requests."
- From https://owasp.org/www-community/attacks/csrf

That said, maybe the `GET` calls for `/canvas/*` could benefit from CSRF protection (at least `canvas/returnFromOAuth` could be considered mutating), but I think there is additional security baked in there with the `state` variable. I'll have to think about that some more.

Reply:
I looked into this a bit more, and the `state` parameter seems to take the place of the CSRF token to prevent attacks here. I will separately research and vet the use of the `state` parameter to ensure we're covered here.
See https://auth0.com/docs/protocols/state-parameters#set-and-compare-state-parameter-values and https://security.stackexchange.com/questions/20187/oauth2-cross-site-request-forgery-and-state-parameter